### PR TITLE
Merge JS package tests into test.yml

### PR DESCRIPTION
**Pull request by mehulkar**

### CODE REVIEW

**Comment by mehulkar** [comment]
Tested in https://github.com/vercel/turbo/pull/4947

**Comment by mehulkar** [comment]
The failing examples test here seems like an issue with the npm registry or something. `pnpm-basic` failed before and now `pnpm-with-svelte`. It's definitely not related to this PR.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -406,6 +414,48 @@ jobs:
...
+
+      - name: Build turborepo CLI from source
+        uses: ./.github/actions/setup-turborepo-environment
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          target: ${{ matrix.os.name }}
+
+      - name: Run tests
+        run: |
+          turbo run check-types test --filter=...[${{ github.event.pull_request.base.sha || 'HEAD^1' }}] --filter=!cli --filter=!turborepo-tests* --color
```
Comment (mehulkar) [inline .github/workflows/test.yml:457]: not making any changes here but if #4911 merges, I used a different filter there. I will consolidate if that is merged.

**Review by chris-olszewski — APPROVED** [review]
One workflow to rule them all

**Pull request merged**